applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1096,8 +1121,17 @@ static void connect(const void *subscriber_id, uint8_t report_id)
 
 		if (cur_sub_prio < new_sub_prio) {
 			LOG_WRN("Force report data unlink");
+			struct report_state *former_rs = rd->linked_rs;
 			rd->linked_rs = NULL;
 			clear_report_data(rd);
+
+			/* Unlink report data from former report sink.
+			 * Instead link an empty report data to it so
+			 * host get a clear report releasing any buttons.
+			 */
+			former_rs->update_needed = true;
+			former_rs->linked_rd = (struct report_data *)&empty_rd;
+			report_send((struct report_data *)&empty_rd, former_rs, true, false);
```

Comment:
`report_send(NULL, former_rs, true, false)`

Reply:
see earlier comment. there is also one more thing to that - we have one place where rd/rs will be derived from respective rs/rd (as needed). lower chance of contradicting something and bugs